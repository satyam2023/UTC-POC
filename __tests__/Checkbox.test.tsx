import {fireEvent, render} from '@testing-library/react-native';
import CustomCheckBox from '../src/components/CheckBox';

describe('Testing the CheckBox', () => {
  const baseStyle = {
    flex: 1,
  };


  test('Making Snapshot', () => {
    const root = render(<CustomCheckBox />);
    expect(root.toJSON()).toMatchSnapshot();
  });

  test('taking first conditional componet', () => {
    const {getByTestId} = render(<CustomCheckBox style={baseStyle} isRectangular={false} />);
    const component = getByTestId('conditional-rendering');
    expect(component.props.style).toEqual(baseStyle);
  });


  test('taking  conditional componet', () => {
    const mckFunc=jest.fn();
  
    const {getByTestId}= render(<CustomCheckBox  isRectangular={true}  onPress={mckFunc} />);
    const component = getByTestId("conditional");
    fireEvent.press(component);
    expect(mckFunc).toHaveBeenCalled();
  });
  
});
