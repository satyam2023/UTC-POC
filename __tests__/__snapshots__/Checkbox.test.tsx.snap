// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CheckBox Making Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "borderColor": "#E6E6E6",
        "borderWidth": 1,
        "height": 17,
        "justifyContent": "center",
        "width": 17,
      },
      undefined,
      {
        "borderRadius": 10,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <View
      testID="conditional-rendering"
    />
  </View>
</View>
`;
